src/modules/module_utils.c

```diff
@@ -413,6 +418,62 @@ void module_strip_punctuation_default(char *buf)
 	module_strip_punctuation_some(buf, "~#$%^&*+=|<>[]_");
 }
 
+gchar *module_multicases_string_replace(
+	const gchar *text,
+	const gchar *pattern,
+	const gchar *replace)
+{
+	GRegex *regex;
+	GError *error = NULL;
+	gchar *result;
+
+	result = g_strdup(text);
+	regex = g_regex_new(pattern, G_REGEX_OPTIMIZE, 0, &error);
+	if (!regex) {
+		DBG("ERROR compiling regular expression: %s.", error->message);
+		g_error_free(error);
+		return result;
+	}
+
+	result = g_regex_replace(regex, text, -1, 0, replace, G_REGEX_MATCH_DEFAULT, NULL);
+
+	g_error_free(error);
+	g_regex_unref(regex);
+							 
+	return result;
+}
+
+char *module_multicases_string(const char *message)
+{
+	gchar *pattern[4];
+	gchar *replace[4];
+	gchar *multicases;
+	gchar *text;
+	guint i;
+
+	assert(message != NULL);
+
+	pattern[0] = "([a-z]+)([A-Z][a-z]+)";
```

Comment:
you probably want to add word boundaries at beginnings and ends of the patterns? (`\b`) to avoid matching in unexpected ways